### PR TITLE
chore: add changelog for v0.1 and v0.2

### PR DESCRIPTION
**Pull request by grjte**
This PR adds a changelog in preparation for the release of v0.2.

### CODE REVIEW

**Comment by grjte** [comment]
We can wait to merge this until the docs PR #152 is done and everything else is ready to go.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Changelog
+
+## 0.2.0 (2023-02-22)
+
+- Added support for named constants (scalars, vectors, and matrices).
+- Added support for intermediate variables in `boundary_constraints` and `transition_constraints` sections (scalars, vectors, and matrices).
```
Comment (tohrnii) [inline CHANGELOG.md:6]: should this be `integrity_constraints` section?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- Added support for binding identifier names to groups of trace columns (in addition to single columns, which were already supported).
+- Added the `$main` and `$aux` built-ins for accessing columns in the execution trace by index.
+- [BREAKING] Replaced the `$rand` built-in for accessing random values with an explicit `random_values` declaration section, enabling use of a custom identifier for the random values array.
+- Added support for binding identifiers to specific random values.
```
Comment (tohrnii) [inline CHANGELOG.md:10]: Should we mention groups as well? Maybe something like:
"Added support for binding identifiers to specific random values or groups of random values."
or do you think that is already covered in the above statement?

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+- Made significant changes to the IR, including:
```
Comment (tohrnii) [inline CHANGELOG.md:11]: Should we say "IR structure" or maybe "IR design" here instead?

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  - Moved the boundary constraints into the algebraic constraint graph.
+  - Made the trace reference more general to support additional trace segments in the future.
+  - Added analysis to differentiate between validity and transition constraints.
```
Comment (tohrnii) [inline CHANGELOG.md:14]: Do you think replacing "analysis" with "ability" would be better?

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  - Added the `Sub` operation and removed the `Neg` operation.
+- [FIX] Fixed a bug in the Winterfell codegen output for auxiliary transition constraints.
```
Comment (tohrnii) [inline CHANGELOG.md:16]: should we be more specific here?

"Fixed a bug in the Winterfell codegen output for auxiliary transition constraints using main trace columns"

**Review by tohrnii — COMMENTED** [review]
Thank you @grjte, looks good to me. I just left a few questions inline.

File: CHANGELOG.md (hunk 4 above)
Comment (grjte) [reply CHANGELOG.md:14]: Hm, no - what I meant is that we now analyze the constraints to determine the type. I could change it to the following if you think it's better: "Added analysis of integrity constraints to identify the constraint type as validity or transition"

File: CHANGELOG.md (hunk 3 above)
Comment (grjte) [reply CHANGELOG.md:11]: I think I'd leave it as it is - we've made changes to both the design and the structure, as well as the interface used for codegen, and "IR" encompasses all of that.

**Comment by grjte** [comment]
Thanks for the review @tohrnii, I've made some of the updates and addressed your other points

File: CHANGELOG.md (hunk 4 above)
Comment (tohrnii) [reply CHANGELOG.md:14]: No in that case I think what you had originally is probably better.

File: CHANGELOG.md (hunk 5 above)
Comment (tohrnii) [reply CHANGELOG.md:16]: Makes sense.

**Review by tohrnii — APPROVED** [review]
Looks good to me. Thanks @grjte.

**Pull request merged**